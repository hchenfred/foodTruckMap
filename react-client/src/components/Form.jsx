import React, {Component} from 'react';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {searchTerm: ''};
    this.handleSubmitForm = this.handleSubmitForm.bind(this);
    this.onPlaceChanged = this.onPlaceChanged.bind(this);
  }

  componentDidMount() {
    this.autocomplete = new google.maps.places.Autocomplete((
      document.getElementById('autocomplete')), {
      types: ['(cities)'],
    });
    this.autocomplete.addListener('place_changed', this.onPlaceChanged);  
  }

  onPlaceChanged() {
    const place = this.autocomplete.getPlace();
    this.setState({searchTerm: place});
  }

  handleSubmitForm(e) {
    e.preventDefault();
    const geometry = this.state.searchTerm.geometry
    if (geometry === null || geometry === undefined) {
      document.querySelector('.dropdown-info').style.display = 'block';
    } else {
      const lat = geometry.location.lat();
      const lng = geometry.location.lng();
      this.props.setCurrLocation({lat, lng});
    }
  }

  render() {
    return (<div className="form-container">
      <form onSubmit={this.handleSubmitForm} className="location-form">
        <div className="row">
        <div className="form-group">
          <input onChange={this.handleInputChange} id="autocomplete" 
          className="form-control formInput" 
          type="text" 
          placeholder="Enter a Location To Search for Stores" 
          required/>
          <div className="dropdown-info">Please use autocompleted address</div>
        </div>
        <div className="form-group">
          <input className="btn btn-info formInput submitBtn btn-block" type="submit" value="Search"/>
        </div>
        </div>
      </form>
    </div>);
  }
}

export default Form;