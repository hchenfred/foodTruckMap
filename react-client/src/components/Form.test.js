import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Form from './Form';

test('Form component should render as expected', () => {
  const component = shallow(<Form />);
  expect(component.find('input').length).toBe(2);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});