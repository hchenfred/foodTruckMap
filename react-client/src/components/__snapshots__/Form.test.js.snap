// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should render as expected 1`] = `
<div
  className="form-container"
>
  <form
    className="location-form"
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <div
        className="form-group"
      >
        <input
          className="form-control formInput"
          id="autocomplete"
          placeholder="Enter an address to search for food trucks"
          required={true}
          type="text"
        />
        <div
          className="dropdown-info"
        >
          Please use autocompleted address
        </div>
      </div>
      <div
        className="form-group"
      >
        <input
          className="btn btn-info formInput submitBtn btn-block"
          type="submit"
          value="Search"
        />
      </div>
    </div>
  </form>
</div>
`;
